src/tabmat/categorical_matrix.py

```diff
@@ -597,38 +597,46 @@ def _cross_sparse(
         L_cols: Optional[np.ndarray],
         R_cols: Optional[np.ndarray],
     ) -> np.ndarray:
-        term_1 = self * d  # multiply will deal with drop_first
+        term_1 = self.multiply(d)  # multiply will deal with drop_first
 
         term_1 = _row_col_indexing(term_1, rows, L_cols)
 
         res = term_1.T.dot(_row_col_indexing(other, rows, R_cols)).A
         return res
 
-    def multiply(self, other) -> sps.csr_matrix:
-        """Element-wise multiplication of each column with other."""
+    def multiply(self, other) -> SparseMatrix:
+        """Element-wise multiplication.
+
+        This assumes that ``other`` is a vector of size self.shape[1].
```

Comment:
This line of docstring seems inconsistent with line 612 below, where the check is performed with ``self.shape[0]`` not on ``self.shape[1]``

Reply:
Thanks! 🤦